Decompose _unsafe_index_put into index_put

jackzhxng opened the pull request:
## Description
Create decomposition of _unsafe_index_put (non-core aten) that turns it into index_put (core aten)

## Testing
Phi3 mini + LoRA model successfully passed `to_edge` after failing due to a non-core aten `unsafe_index_put` getting introduced in a decomposition during joint graph calculations.

larryliu0820 commented:
Can you fix the CI job? Need to fix `test_decomp.py::HasDecompTest::test_has_decomposition`

pianpwk commented on torch/_decomp/__init__.py:
```diff
@@ -328,6 +328,7 @@ def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
             aten.huber_loss,
             aten.huber_loss_backward,
             aten.im2col,
+            aten._unsafe_index_put,
```
nit: put this later in the list next to `aten._unsafe_index`, so it's in order

Commit:
Decompose _unsafe_index_put into index_put (#133365)

Summary:
## Description
Create decomposition of _unsafe_index_put (non-core aten) that turns it into index_put (core aten)

## Testing
Phi3 mini + LoRA model successfully passed `to_edge` after failing due to a non-core aten `unsafe_index_put` getting introduced in a decomposition during joint graph calculations.


Differential Revision: D61234115

Pulled By: dvorjackz
